### PR TITLE
Support cassandracluster in paasta status and start/stop/restart

### PR DESCRIPTION
**Pull request by keymone**
Refactored flink related code to be more generic and added cassandracluster instance type support.

### CODE REVIEW

File: paasta_tools/cli/cmds/start_stop_restart.py [hunk 1]
```diff
@@ -253,6 +253,7 @@ def paasta_start_or_stop(args, desired_state):
                     soa_dir=soa_dir,
                     load_deployments=False,
                 )
+                print(service_config)
```
Comment (vkhromov) [inline paasta_tools/cli/cmds/start_stop_restart.py]: rm

File: paasta_tools/cli/cmds/status.py [hunk 2]
```diff
@@ -1290,13 +1292,17 @@ def paasta_status(args,) -> int:
     clusters_services_instances = apply_args_filters(args)
     for cluster, service_instances in clusters_services_instances.items():
         for service, instances in service_instances.items():
+            print(cluster, service, instances)
```
Comment (vkhromov) [inline paasta_tools/cli/cmds/status.py]: rm

File: paasta_tools/api/views/instance.py [hunk 3]
```diff
@@ -155,28 +155,16 @@ def adhoc_instance_status(
...
-) -> Optional[Mapping[str, Any]]:
-    status: Optional[Mapping[str, Any]] = None
+def kubernetes_cr_status(cr_id: dict, verbose: int) -> Optional[Mapping[str, Any]]:
     client = settings.kubernetes_client
     if client is not None:
-        status = flink_tools.get_flink_status(
-            kube_client=client, service=service, instance=instance
-        )
-    return status
+        return kubernetes_tools.get_cr_status(kube_client=client, cr_id=cr_id)
```
Comment (Bronek) [inline paasta_tools/api/views/instance.py]: Missing "return None" in "else" execution path (see mypy)

File: paasta_tools/api/views/instance.py [hunk 4]
```diff
@@ -155,28 +155,16 @@ def adhoc_instance_status(
...
-) -> Optional[Mapping[str, Any]]:
-    metadata: Optional[Mapping[str, Any]] = None
+def kubernetes_cr_metadata(cr_id: dict, verbose: int) -> Optional[Mapping[str, Any]]:
     client = settings.kubernetes_client
     if client is not None:
-        metadata = flink_tools.get_flink_metadata(
-            kube_client=client, service=service, instance=instance
-        )
-    return metadata
+        return kubernetes_tools.get_cr_metadata(kube_client=client, cr_id=cr_id)
```
Comment (Bronek) [inline paasta_tools/api/views/instance.py]: Missing "return None" in "else" execution path (see mypy)

**Comment by Bronek** [comment]
There are two more CI failures:

```
Reorder python imports...................................................Failed
hookid: reorder-python-imports
Files were modified by this hook. Additional output:
Reordering imports in paasta_tools/cli/cmds/status.py
Reordering imports in paasta_tools/flink_tools.py
Reordering imports in paasta_tools/cassandracluster_tools.py
Reordering imports in paasta_tools/api/views/instance.py
. . .
flake8...................................................................Failed
hookid: flake8
paasta_tools/cassandracluster_tools.py:13:1: F401 'typing.Any' imported but unused
paasta_tools/cassandracluster_tools.py:19:1: F401 'kubernetes.client.rest.ApiException' imported but unused
paasta_tools/cassandracluster_tools.py:21:1: F401 'paasta_tools.kubernetes_tools.get_cr_metadata' imported but unused
paasta_tools/cassandracluster_tools.py:22:1: F401 'paasta_tools.kubernetes_tools.get_cr_status' imported but unused
paasta_tools/cassandracluster_tools.py:24:1: F401 'paasta_tools.kubernetes_tools.KubeClient' imported but unused
paasta_tools/flink_tools.py:13:1: F401 'typing.Any' imported but unused
paasta_tools/flink_tools.py:19:1: F401 'kubernetes.client.rest.ApiException' imported but unused
paasta_tools/flink_tools.py:22:1: F401 'paasta_tools.kubernetes_tools.get_cr_metadata' imported but unused
paasta_tools/flink_tools.py:23:1: F401 'paasta_tools.kubernetes_tools.get_cr_status' imported but unused
paasta_tools/flink_tools.py:25:1: F401 'paasta_tools.kubernetes_tools.KubeClient' imported but unused
```

**Comment by Bronek** [comment]
@keymone I plan to merge #2427 soon and you will have to resolve some conflicts when I do. Do you want me to wait for you to merge first?

**Comment by keymone** [comment]
@Bronek ack, i'll rebase

**Comment by Bronek** [comment]
There are some failures in the CI from black, can you please fix them?

File: paasta_tools/api/views/instance.py [hunk 5]
```diff
@@ -90,6 +90,10 @@
 log = logging.getLogger(__name__)
 
 
+INSTANCE_TYPES_K8S = {"flink", "cassandracluster"}
```
Comment (vkhromov) [inline paasta_tools/api/views/instance.py]: there is `INSTANCE_TYPES` in `paasta_tools/utils.py`, may be move `INSTANCE_TYPES_K8S ` there?

Comment (keymone) [reply paasta_tools/api/views/instance.py]: will do, thanks!

**Pull request merged**